src/ui/win32/bindings/GridTextColumnBinding.cpp

```diff
@@ -70,7 +72,26 @@ static LRESULT CALLBACK EditBoxProc(HWND hwnd, UINT nMsg, WPARAM wParam, LPARAM
                 return GridBinding::CloseIPE(hwnd, pInfo);
             }
 
+            // BizHawk specific hack, seems BizHawk does not have WM_CHAR messages sent, so text can't be typed
+            // It's unknown why this occurs, it's likely something to do with it using IsDialogMessage more properly in
+            // its message loop and something is likely wrong with the way edit windows are handled in RAIntegration.
+            // Until that is figured out, this hack seems to suffice
+            if (ra::services::ServiceLocator::Get<ra::data::context::EmulatorContext>().GetEmulatorId() ==
+                EmulatorID::BizHawk)
+            {
+                MSG msg = {hwnd, nMsg, wParam, lParam};
+                TranslateMessage(&msg);
+                PeekMessage(&msg, hwnd, 0, 0, PM_REMOVE);
+                if (msg.message == nMsg) // translation didn't happen, go to original proc
+                    break;
+                DispatchMessage(&msg);
+                return 0;
+            }
+
             break;
+
+        case WM_GETDLGCODE:
+            return DLGC_WANTALLKEYS;
```

Comment:
Correct, [IsDialogMessage sends WM_GETDLGCODE messages to the dialog box procedure to determine which keys should be processed.](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-isdialogmessagew). Interesting it gets sent to RALibretro anyways under certain conditions. MS does mention DestroyWindow [flushes the thread message queue](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-destroywindow), so it can probably be assumed it does this properly with IsDialogMessage.